src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,86 @@
+package duke.task;
+
+import duke.Message;
+
+public class Task {
+    private boolean isDone;
+    private String description;
+    private final Types type;
+
+    public enum Types {
```

Comment:
Umm may you could explain me why do you want to store these types as `enums`. If you are inheriting everything from either `TimedTask` or `Task` then whats the need of these types?.
I would recommend this approach where lets say `Event` inherits from `TimedTask` and has all the functionalities of `Event`. With such an approach the class `Task` or `TimedTask` does not have to know that Event class exists (separation of concerns)

Reply:
I tend to use enums to store 'short strings' like commands and task names. the advantage I see is that you can still get the string in uppercase using name() method and lowercase by overwriting the toString() method. Also, from my understanding, switch statements become less computationally intensive with enums because you can use the typeOf() method to convert the string to an enum and switch based on the enum (which is an int as opposed to a string). Also, my Event does inherit from TimedTask. Given that 'Deadline' and 'Event' are similar except for their preposition, I used an abstract parent class TimedTask to implement the shared methods.